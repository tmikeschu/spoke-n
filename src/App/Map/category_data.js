const categoryIcons = {
  stay: "https://maps.google.com/mapfiles/ms/icons/blue.png",
  checkout: "https://maps.google.com/mapfiles/ms/icons/green.png",
  avoid: "https://maps.google.com/mapfiles/ms/icons/red.png",
  bike_shop: "https://maps.google.com/mapfiles/ms/icons/mechanic.png",
  other: "https://maps.google.com/mapfiles/ms/icons/purple.png",
  endOfDay: "https://maps.google.com/mapfiles/ms/icons/flag.png",
  cycling: "https://maps.google.com/mapfiles/ms/icons/cycling.png",
}

const categoryLabels = {
  "stay": "Places to Stay",
  "checkout": "Cool Spots",
  "avoid": "Places to Avoid",
  "bike_shop": "Bike Shops",
  "other": "Other",
  "": "All",
  "DISPLAYNONE": "None"
};

export { categoryIcons, categoryLabels }

