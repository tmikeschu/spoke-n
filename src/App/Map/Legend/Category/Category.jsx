import React, { Component } from 'react';
import { categoryIcons, categoryLabels } from '../../category_data';

export default class Category extends Component {
  categoryIcons() {
    return categoryIcons;
  }

  categoryLabels() {
    return categoryLabels;
  }

  render() {
    const category = this.props.category;

    return(
      <li style={{display: this.props.tabIndex === 0 ? 'none' : 'block'}}>
        <span>
          <img 
            src={this.categoryIcons()[category]} 
            alt={category}
          />
        </span> 
        â†’ { this.categoryLabels()[category] }
      </li>
    );
  }
}
