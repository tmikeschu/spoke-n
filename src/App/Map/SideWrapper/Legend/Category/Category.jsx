import React from "react"
import PropTypes from "prop-types"
import Emoji from "../../../Emoji"
import { categoryIcons, categoryLabels } from "../../../category_data"

const Category = ({ category }) => (
  <li>
    <Emoji emoji={categoryIcons[category]} />
    â†’ {categoryLabels[category]}
  </li>
)

Category.propTypes = {
  category: PropTypes.string.isRequired,
}

export default Category
