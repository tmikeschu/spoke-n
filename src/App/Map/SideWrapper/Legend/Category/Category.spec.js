import React from 'react'
import { shallow } from "enzyme"
import Category from "./Category"
import { categoryIcons, categoryLabels } from '../../../category_data'

describe("<Category />", () => {
  const category = shallow(<Category category="stay" />)

  it("renders without crashing", () => {
    expect(category).toBeTruthy()
  })

  it("renders the correct content", () => {
    expect(category.equals(
      <li>
        <span>
          <img 
            src={categoryIcons["stay"]} 
            alt="stay"
          />
        </span> 
        â†’ { categoryLabels["stay"] }
      </li>
    )).toBeTruthy()
  })

  describe("snapshot", () => {
    it("is valid", () => {
      expect(category.getNodes()).toMatchSnapshot()
    })
  })
})
