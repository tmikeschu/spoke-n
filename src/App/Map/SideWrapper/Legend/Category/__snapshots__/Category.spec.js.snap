// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Category /> snapshot is valid 1`] = `
<li>
  <Emoji
    emoji="🏡"
  />
  → 
  Places to Stay
</li>
`;
